app/src/main/java/com/monstarlab/core/mapper/Mapper.kt

```diff
@@ -0,0 +1,9 @@
+package com.monstarlab.core.mapper
+
+/**
+ * Created by aslam on 08,March,2023
+ */
+interface Mapper<I, O> {
+    fun map(input: I): O
+}
+
```

Comment:
From my experience using the Mapper interface doesn't provide any value to it and it the end is a bit annoying.
In this template we ended up using extensions functions on the DTO to do the mapping, which is more straightforward IMO.

You can check [ResourceDTO](https://github.com/monstar-lab-oss/android-template/blob/main/app/src/main/java/com/monstarlab/features/resources/data/api/dtos/ResourceDTO.kt#L18) for the example

Reply:
This approach using the interface and mapper implementation is more flexible and reusable. It enables you to define a generic mapping interface and multiple mapper implementations for various types of data. This approach is well-suited for larger projects that necessitate extensive data mapping and may have multiple data sources or layers.

In contrast, the approach you are proposing, which employs extension functions, is more succinct and user-friendly. It provides a rapid method for mapping data without creating supplementary classes or implementing an interface. This approach is suitable for smaller projects or components of larger projects where data mapping is not the primary focus.

Should I consider changing the implementation? What do you think?

Reply:
I would say in most scenarios having an extension function will be enough. 
Yes, you won't be able to inject the extension function, but that's about the only limitation I see, it doesn't remove any flexibility or reusability.

Extensions are as reusable as the Mapper interface. Having this much boilerplate to just enable Mapping in the repository does not appeal to me. So in the end yes, I think we should go with the extension function for the purpose of this template. 

As soon as the developer sees the need in having such a structure and the Mapping is very complex it deserves its own component, then yes extension function doesn't make sense and  this of course is the way to go